src/wmtk/TriMeshOperationExecutor.cpp

```diff
@@ -128,94 +121,54 @@ void TriMesh::TriMeshOperationExecutor::glue_ear_to_face(
     }
 }
 
-void TriMesh::TriMeshOperationExecutor::merge(const long &new_vid)
+void TriMesh::TriMeshOperationExecutor::merge(const long& new_vid)
 {
-    simplices_to_delete[0].push_back(end_point_vids[0]);
-    simplices_to_delete[0].push_back(end_point_vids[1]);
-    simplices_to_delete[1].push_back(E_AB_id);
-    for (int face_index = 0; face_index < FaceDatas.size(); face_index++) {
-        if (FaceDatas[face_index].ears[0].fid < 0 && FaceDatas[face_index].ears[1].fid < 0) {
+    simplices_to_delete[0].push_back(m_incident_vids[0]);
+    simplices_to_delete[0].push_back(m_incident_vids[1]);
+    simplices_to_delete[1].push_back(m_operating_edge_id);
+    for (int face_index = 0; face_index < m_incident_face_datas.size(); face_index++) {
+        if (m_incident_face_datas[face_index].ears[0].fid < 0 &&
+            m_incident_face_datas[face_index].ears[1].fid < 0) {
             return; // TODO: throw exception, should be detected by link condition
         }
 
-        if (FaceDatas[face_index].ears[0].fid == FaceDatas[face_index].ears[1].fid) {
+        if (m_incident_face_datas[face_index].ears[0].fid ==
+            m_incident_face_datas[face_index].ears[1].fid) {
             return; // TODO: throw exception, non-manifold
         }
 
         // change VF for V_new,V_C
         long& vf_new = vf_accessor.scalar_attribute(new_vid);
-        long& vf_c = vf_accessor.scalar_attribute(FaceDatas[face_index].V_C_id);
-        const long f_ear_l = FaceDatas[face_index].ears[0].fid;
-        const long f_ear_r = FaceDatas[face_index].ears[1].fid;
+        long& vf_c = vf_accessor.scalar_attribute(m_incident_face_datas[face_index].opposite_vid);
+        const long f_ear_l = m_incident_face_datas[face_index].ears[0].fid;
```

Comment:
i'd rather we did `auto& incident_face_data = m_incident_face_datas[face_index];` to reduce array indexing. it's a habit from teh old codebase we shoud really avoid